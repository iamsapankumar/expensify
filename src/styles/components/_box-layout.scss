.box-layout {
   align-items: center;
   background: url('/images/bg.jpg');
   background-size: cover;
   display: flex;
   height: 100vh;
   justify-content: center;
   width: 100vw; 

   &__box {
       background: fade-out($color: white, $amount: .15);
       border-radius: 5px;
       padding: $l-size $xl-size;
       text-align: center;
       width: 25rem;

   }
   &__title {
       margin: 0 0 $m-size 0;
       line-height: 1;
   }
}