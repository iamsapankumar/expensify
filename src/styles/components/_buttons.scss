.button {
    background: $blue;
    border-radius: 5px;
    border: none;
    color: white;
    display: inline-block;
    font-weight: 300;
    font-size: $font-size-large;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;

    &--link {
        background: none;
    }
    &--remove {
        @extend .button;
        background: gray;
    }
}