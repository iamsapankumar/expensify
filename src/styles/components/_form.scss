.form {
    display: flex;
    flex-direction: column;
    > * {
        margin-bottom: $m-size;
    }

    &__error {
      margin: 0 0 $m-size $s-size;
      font-style: italic;
      font-weight: bolder;
    }
}