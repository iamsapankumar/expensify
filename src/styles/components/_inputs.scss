.text-input {
    border: 1px solid #cacccd;
    height: 4.5rem;
    font-size: $font-size-large;
    font-weight: 300;
    outline: none;
    padding: $s-size;
}
.select {
    @extend .text-input;
    background: none;
}
.textarea {
    @extend .text-input;
    height: 10rem;
}