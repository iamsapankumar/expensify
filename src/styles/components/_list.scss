.list-header {
    background: $light-gray;
    border: darken($light-gray, 10%);
    color: $grey;
    display: flex;
    justify-content: space-between;
    padding: $s-size $m-size;
}
.list-item {
    border-bottom: 1px solid darken($light-gray, 30%);
    color: $dark-gray;
    display: flex;
    flex-direction: column;
    padding: $s-size;
    text-decoration: none;
    transition: background .3s ease;

    &:hover {
        background: darken($light-gray, 10%);
    }
    @media (min-width: $desktop-breakpoint) {
        align-items: center;
        flex-direction: row;
        justify-content: space-between;
        padding: $m-size;
    }
    &__title {
      margin: 0;
      word-break: break-all;
    }
    &__sub-title {
      color: $grey;
      font-size: 1.4rem;
    }
    &__data {
      margin-left: $s-size 0 0 0;

      @media(min-width: $desktop-breakpoint) {
          margin: 0;
          padding-left: $s-size;
      }
    }
    &--message {
        align-items: center;
        color: $grey;
        justify-content: center;
        padding: $m-size;
        &:hover {
            background: none;
        }
    }
}
.list-body {
    margin-bottom: $m-size;
    @media(min-width: $desktop-breakpoint) {
        margin-bottom: $xl-size;
    }
}