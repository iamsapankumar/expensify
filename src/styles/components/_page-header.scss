.page-header {
    background-color: #f7f7f7;
    margin-bottom: $l-size;
    padding: $l-size 0;

    &__actions {
        margin-top: $l-size;
    }

    &__title {
      font-weight: 300;
      margin: 0;

      span {
          font-weight: bold;
      }
    }
}