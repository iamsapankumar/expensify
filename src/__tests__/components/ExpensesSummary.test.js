import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should render correctly ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary
        expenseCount={1}
        expenseTotal={235}
    />);
    expect(wrapper).toMatchSnapshot();
});
test('Should render correctly ExpensesSummary with expenses', () => {
    const wrapper = shallow(<ExpensesSummary
        expenseCount={3}
        expenseTotal={10456}
    />);
    expect(wrapper).toMatchSnapshot();
});




