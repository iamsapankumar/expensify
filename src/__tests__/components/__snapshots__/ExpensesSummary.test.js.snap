// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`Should render correctly ExpensesSummary with expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $104.56
  </h1>
</div>
`;
